site/docs/id/guide/filter-queries.md

```diff
@@ -350,8 +350,23 @@ Selain itu, kamu juga akan diberikan pesan error yang lebih bermanfaat.
 Validasi filter query hanya dilakukan sekali, ketika bot diinisialisasi dan `bot.on()` dipanggil.
 
 Ketika dimulai, grammY menurunkan _function predicate_ dari filter query dengan cara memecahnya menjadi beberapa komponen query.
-Setiap komponen akan di-map ke sebuah function yang mengerjakan satu pemeriksaan `in`, atau dua pemeriksaan jika komponen tersebut diabaikan dan dua value perlu dilakukan pemeriksaan.
+Setiap komponen akan di-map ke sebuah function yang mengerjakan satu pemeriksaan truthiness untuk properti objek, atau dua pemeriksaan jika komponen tersebut diabaikan dan dua value perlu dilakukan pemeriksaan.
 Function-function ini kemudian disatukan untuk membentuk sebuah predicate yang akan memeriksa sebanyak mungkin value yang relevan untuk query, tanpa melakukan proses perulangan terhadap key object `Update`.
 
 Sistem ini menggunakan lebih sedikit operasi dibandingkan dengan beberapa library lainnya, dimana dibutuhkan beberapa pengecekan array ketika melakukan routing update.
 Ini membuat sistem filter query grammY selain lebih unggul juga jauh lebih cepat.
+
+### Type Safety
+
+Sebagaimana yang disebutkan diatas, filter queries secara otomatis akan mempersempit properti tertentu pada objek context dan menghasilkan satu atau lebih predicate dalam fungsi predicate type di TypeScript.
+Secara umum, kamu dapat mempercayai type inference itu ia selalu bekerja dengan benar.
+Jika suatu properti type infer-nya tersedia, kamu dapat dapat menggunakan type tersebut dengan aman.
+Jika suatu properti type infer-nya hilang, berarti ada kasus tertentu yang membuat type-nya hilang.
+Bukan ide yang baik untuk melakukan casting type dengan operator `!`.
+
+> Mungkin ini kurang jelas tentang apa maksud kasus tersebut.
+> Jangan ragu untuk bertanya di [obrolan grup](https://t.me/grammyjs) jika kamu tidak memahaminya.
+
+Membuat type ini merumitkan.
+Banyak sekali pengetahuan tentang API Bot yang masuk pada bagian grammY ini.
+Apabila kamu ingin lebih memahami tentang pendekatan dasar bagaimana type ini dibuat, berikut ada [pembahasannya di Youtube](https://youtu.be/ZvT_xexjnMk) yang mungkin bisa kamu tonton.
```

Comment:
@dzikrl perhaps it should be mentioned that the presentation on YouTube is in English (as a warning), what do you think?

Reply:
> @dzikrl perhaps it should be mentioned that the presentation on YouTube is in English (as a warning), what do you think?

It's okay to not mention it. In [grammY introduction](https://grammy.dev/id/guide/introduction#persiapan-untuk-memulai) about "Not Learning How to Code" on Indonesian translation. There is also a link Youtube video that in English too. But, @quadratz improve that with a some note.

cc : @quadratz?